src/Mailer/Transport/SendgridTransport.php

```diff
@@ -93,8 +88,13 @@ protected function _send($message)
             'headers' => ['Authorization' => 'Bearer ' . $this->transportConfig['api_key']]
         ];
         $response = $this->http->post('/api/mail.send.json', $message, $options);
-        if (!$response) {
-            throw new SocketException($response->code);
+        if ($response->code !== 200) {
+            throw new SocketException(
+                'SendGrid error ' .
+                $response->getStatusCode() . ' ' .
+                $response->getReasonPhrase() . ': ' .
+                implode('; ', $response->json['errors'])
+            );
```

Comment:
Accessing `$response->code` worked fine in the original code, but when I tried to use `$response->reasonPhrase` I didn't get anything, so I switched to using the corresponding getters. That worked in both cases.

A dump of `$response->json` looks like this for success:
```
Array
(
    [message] => success
)
```
And this for failure:
```
Array
(
    [errors] => Array
        (
            [0] => Missing email body
        )

    [message] => error
)
```